sql/database.go

```diff
@@ -1165,7 +1272,13 @@ func (c *sqliteConn) Exec(query string, encoder Encoder, decoder Decoder) (int,
 			c.db.latency.WithLabelValues(query).Observe(float64(time.Since(start)))
 		}()
 	}
-	return exec(c.conn, query, encoder, decoder)
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	conn := c.ensureConn()
+	defer func() {
+		c.timer.Reset(c.db.connIdleTimeout)
+	}()
```

Comment:
Does `c`, `c.timer`, `c.db` or `c.db.connIdleTimeout` change between this line and `exec` returning? If not why not just

```suggestion
	defer c.timer.Reset(c.db.connIdleTimeout)
```

Reply:
Fixed